hana/lib/HANAService.js

```diff
@@ -60,7 +60,12 @@ class HANAService extends SQLService {
           HANAVERSION = dbc.server.major
           return dbc
         } catch (err) {
-          if (!isMultitenant || err.code !== 10) throw err
+          if (isMultitenant) {
+            // REVISIT: throw the error and break retry loop
+            // Stop trying when the tenant does not exist or is rate limited
+            if (err.status == 404 || err.status == 429)
+              return new Promise(() => { })
```

Comment:
I think we need to have a timeout here as otherwise we have unresolving promises?!

Reply:
```suggestion
              return new Promise(function (_, reject) { setTimeout(() => reject(e)), this._config.acquireTimeoutMillis})
```